lib/ts/recipe/session/cookieAndHeaders.ts

```diff
@@ -205,3 +206,97 @@ export function setCookie(
 export function getAuthModeFromHeader(req: BaseRequest): string | undefined {
     return req.getHeaderValue(authModeHeaderKey)?.toLowerCase();
 }
+
+/**
+ *  This function addresses an edge case where changing the cookieDomain config on the server can
+ *  lead to session integrity issues. For instance, if the API server URL is 'api.example.com'
+ *  with a cookie domain of '.example.com', and the server updates the cookie domain to 'api.example.com',
+ *  the client may retain cookies with both '.example.com' and 'api.example.com' domains.
+ *
+ *  Consequently, if the server chooses the older cookie, session invalidation occurs, potentially
+ *  resulting in an infinite refresh loop. To fix this, users are asked to specify "olderCookieDomain" in
+ *  the config.
+ *
+ * This function checks for multiple cookies with the same name and clears the cookies for the older domain
+ */
+export function clearSessionCookiesFromOlderCookieDomain({
+    req,
+    res,
+    config,
+    userContext,
+}: {
+    req: BaseRequest;
+    res: BaseResponse;
+    config: TypeNormalisedInput;
+    userContext: UserContext;
+}): boolean {
+    let didClearCookies = false;
+
+    if (config.olderCookieDomain === undefined) {
+        return didClearCookies;
+    }
+
+    const tokenTypes: TokenType[] = ["access", "refresh"];
+    for (const token of tokenTypes) {
+        if (hasMultipleCookiesForTokenType(req, token)) {
+            logDebugMessage(
+                `clearSessionCookiesFromOlderCookieDomain: Clearing duplicate ${token} cookie with domain ${config.olderCookieDomain}`
+            );
+            setToken(
+                { ...config, cookieDomain: config.olderCookieDomain },
+                res,
+                token,
+                "",
+                0,
+                "cookie",
+                req,
+                userContext
+            );
+            didClearCookies = true;
+        }
+    }
+
+    if (didClearCookies) {
+        throw new SessionError({
+            message:
+                "The request contains multiple session cookies. We are clearing the cookie from olderCookieDomain. Session will be refreshed in the next refresh call.",
+            type: SessionError.CLEAR_DUPLICATE_SESSION_COOKIES,
+        });
+    }
+    return didClearCookies;
```

Comment:
We do not need to return anything here, since it can only be false and we ignore the return value anyway.

Reply:
Agreed. Fixed it.